Automattic/vip-go-mu-plugins
# CANTINA-855 - Account for all indexable post types and statuses when determining the number of shards for index

## Description
In `filter__ep_default_index_number_of_shards()`, we assume the only indexed post types are `post` with a post status of `publish`, which is incorrect. The default is still set at 1, but we will use the hooks `ep_post_mapping` and `ep_user_mapping` to determine if it needs an increase.

## Changelog Description

### Plugin Updated: Enterprise Search

Account for all indexable post types and statuses when determining the number of shards to index

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test
1) Create a large site with only one 'post' and over a million of a public CPT
2) Add a new index version and check that the shards are set at 4
